task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -689,7 +692,11 @@ def _handle_request(self, msg, log):
             self._task_end_time_monotonic = time.monotonic()
         elif isinstance(msg, GetConnection):
             conn = self.client.connections.get(msg.conn_id)
-            resp = conn.model_dump_json(exclude_unset=True).encode()
+            if isinstance(conn, ConnectionResponse):
+                conn_result = ConnectionResult.from_conn_response(conn)
+                resp = conn_result.model_dump_json(exclude_unset=True).encode()
+            elif isinstance(conn, ErrorResponse):
+                resp = conn.model_dump_json().encode()
```

Comment:
It can just be one of the two right? `if / else` better?

Reply:
I wanted to ensure that it is `ErrorResponse`, but if we have other type of msgs, I will replace `elif` with `else`